quantum/pointing_device/pointing_device_auto_mouse.c

```diff
@@ -0,0 +1,216 @@
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "pointing_device_auto_mouse.h"
+
+/* needed variables for tracking auto mouse */
+static auto_mouse_context_t auto_mouse = {.config.layer = (uint8_t)AUTO_MOUSE_DEFAULT_LAYER};
+
+/* Handle checking if a keycode is a mouse button */
+static bool is_mouse_record(uint16_t keycode, keyrecord_t* record) {
+    // allow for keyboard to hook in and override if need be
+    if(is_mouse_record_kb(keycode, record)) return true;
+    switch(keycode) {
+        case KC_MS_UP ... KC_MS_WH_RIGHT:
+            return true;
+    }
+    return false;
+}
+
+/* toggle mouse layer setting (disabling automouse layer changes) */
+void toggle_mouse_layer(void) {
+    auto_mouse.status.layer_toggled ^= 1;
+}
+
+/* Allow for getting auto mouse state */
+bool get_auto_mouse_state(void) {
+    return auto_mouse.config.enabled;
+}
+
+/* Allow for setting auto mouse enable state */
+void set_auto_mouse_state(bool state) {
+    // skip if already set
+    if(auto_mouse.config.enabled == state) return;
+    auto_mouse.config.enabled = state;
+    // reset settings on state change
+    auto_mouse.status.mouse_key_tracker = 0;
+    auto_mouse.timer.active             = 0;
+    auto_mouse.timer.delay              = 0;
+    auto_mouse.status.layer_toggled     = false;
+}
+
+/* set which layer will act as the mouse layer */
+void set_auto_mouse_layer(uint8_t LAYER) {
+    auto_mouse.config.layer = LAYER;
+}
+
+/* Update the auto mouse if there is mouse motion on the base layer */
+void pointing_device_task_auto_mouse(report_mouse_t mouse_report) {
+    // skip if disabled or layer not set
+    if (!auto_mouse.config.enabled) return;
+    // skip if oneshot mouse layer is active
+#ifndef NO_ACTION_ONESHOT
+    if(get_oneshot_layer() == auto_mouse.config.layer) return;
+#endif
+    // Check for mouse movement and update
+    if ((mouse_report.x != 0 || mouse_report.y !=0)) {
+        auto_mouse.timer.active = timer_read();
+        if(
+            !layer_state_is(auto_mouse.config.layer) && 
+            timer_elapsed(auto_mouse.timer.delay) > AUTO_MOUSE_DELAY
+          ){
+            layer_on(auto_mouse.config.layer);
+            auto_mouse.timer.delay = 0;
+        }
+    } else if (
+        timer_elapsed(auto_mouse.timer.active) > AUTO_MOUSE_TIME &&
+        layer_state_is(auto_mouse.config.layer) &&
+        !auto_mouse.status.mouse_key_tracker &&
+        !auto_mouse.status.layer_toggled
```

Comment:
This is a bit more readable, IMO. 
```suggestion
    } else if (timer_elapsed(auto_mouse.timer.active) > AUTO_MOUSE_TIME && layer_state_is(auto_mouse.config.layer) &&
        !auto_mouse.status.mouse_key_tracker && !auto_mouse.status.layer_toggled
```

Reply:
Whoops changed things out of sync will add this manually.
Your way seems fine to me.